src/LiveComponent/src/EventListener/QueryStringInitializeSubscriber.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadata;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadataFactory;
+use Symfony\UX\LiveComponent\Util\LiveControllerAttributesCreator;
+use Symfony\UX\LiveComponent\Util\QueryStringPropsExtractor;
+use Symfony\UX\TwigComponent\Event\PreMountEvent;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+class QueryStringInitializeSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly RequestStack $requestStack,
+        private readonly LiveComponentMetadataFactory $metadataFactory,
+        private readonly QueryStringPropsExtractor $queryStringPropsExtractor,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            PreMountEvent::class => 'onPreMount',
+        ];
+    }
+
+    public function onPreMount(PreMountEvent $event): void
+    {
+        $metadata = new LiveComponentMetadata(
+            $event->getMetadata(),
+            $this->metadataFactory->createPropMetadatas(new \ReflectionClass($event->getComponent()::class))
```

Comment:
Could it just be this?

```php
$metadata = $this->metadataFactory->getMetadata($event->getMetadata()->getName());
```

Reply:
I hoped that would have worked, but it breaks the hydrator tests because the tested component is mounted manually and thus is not registered in the class map:

```
InvalidArgumentException: Unknown component "__testing". And no matching anonymous component template was found.
```

Is there something to fix here?

Reply:
Is this only a tests problem ? If yes let's find a way to resolve it :)

Reply:
What is blocking precisely ? Can i help you with something ?